<?xml version="1.0" encoding="utf-8"?>

<launch>

  <arg name="map_frame" default="/map" />
  <arg name="lidar_frame" default="/lidar" />
  <arg name="cad_frame" default="/marker2" />
  <arg name="localization_topic" default="/icp_pose" />
  <arg name="scan_topic" default="/corrected_scan" />
  <arg name="distance_threshold" default="0.2" />

  <!-- urdf model -->
  <param name="robot_description"
         textfile="$(find cpt_localization)/launch/mesh_model.urdf" />
  <node pkg="robot_state_publisher" type="state_publisher" name="robot_state_publisher" />

  <!-- cadify node -->
  <node name="localizer_node"
        pkg="cpt_localization"
        type="mesh_localizer_ros_node"
        output="screen"
        respawn="true">
    <param name="off_model"
           value="$(find cpt_localization)/resources/cla_updated.off" />
    <param name="stl_model" value="$(find cpt_localization)/resources/model.dae" />
    <param name="map_frame" value="$(arg map_frame)" />
    <param name="cad_frame" value="$(arg cad_frame)" />
    <param name="lidar_frame" value="$(arg lidar_frame)" />
    <param name="localization_topic" value="$(arg localization_topic)" />
    <param name="scan_topic" value="$(arg scan_topic)" />
    <param name="distance_threshold" value="$(arg distance_threshold)" />

  </node>
</launch>
